### PR TITLE
fix code quality issues

### PR DESCRIPTION
**Pull request by withshubh**
#### Description
This PR fixes a few issues that were affecting the code quality.

#### Summary of changes
- Error classes should inherit from `RuntimeError` instead of `Exception`
- Fix parameters being passed to `expand_path`

### CODE REVIEW

**Comment by withshubh** [comment]
@dsander :wave: Please review this PR! :sparkles:

**Comment by withshubh** [comment]
Hi @bkeepers Please review this PR :sparkles:

File: lib/dotenv/parser.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@
 require "dotenv/substitutions/command" if RUBY_VERSION > "1.8.7"
 
 module Dotenv
-  class FormatError < SyntaxError; end
+  class FormatError < RuntimeError; end
```
Comment (bkeepers) [inline lib/dotenv/parser.rb:5]: Is this causing issues, or what is the motivation to switch from SyntaxError to RuntimeError?

**Review by bkeepers — CHANGES_REQUESTED** [review]
Hi @withshubh, thanks for the pull request! Can you explain why these are considered code quality issues?

**Comment by bkeepers** [comment]
@withshubh I’m going to close this since I didn’t get a reply, but let me know if you want to discuss and have it reopened.

**Pull request closed without merging**